reward/staking_info.go

```diff
@@ -174,6 +200,77 @@ func (s *StakingInfo) DecodeRLP(st *rlp.Stream) error {
 	return nil
 }
 
+func (s *StakingInfo) GetConsolidatedStakingInfo() *ConsolidatedStakingInfo {
+	c := &ConsolidatedStakingInfo{
+		nodes:     make([]consolidatedNode, 0),
+		nodeIndex: make(map[common.Address]int),
+	}
+
+	rewardIndex := make(map[common.Address]int) // temporarily map rewardAddr -> index in []nodes
+
+	for j := 0; j < len(s.CouncilNodeAddrs); j++ {
+		var (
+			nodeAddr      = s.CouncilNodeAddrs[j]
+			stakingAddr   = s.CouncilStakingAddrs[j]
+			rewardAddr    = s.CouncilRewardAddrs[j]
+			stakingAmount = s.CouncilStakingAmounts[j]
+		)
+		if idx, ok := rewardIndex[rewardAddr]; !ok {
+			c.nodes = append(c.nodes, consolidatedNode{
+				NodeAddrs:     []common.Address{nodeAddr},
+				StakingAddrs:  []common.Address{stakingAddr},
+				RewardAddr:    rewardAddr,
+				StakingAmount: stakingAmount,
+			})
+			c.nodeIndex[nodeAddr] = len(c.nodes) - 1 // point to new element
+			rewardIndex[rewardAddr] = len(c.nodes) - 1
+		} else {
+			c.nodes[idx].NodeAddrs = append(c.nodes[idx].NodeAddrs, nodeAddr)
+			c.nodes[idx].StakingAddrs = append(c.nodes[idx].StakingAddrs, stakingAddr)
+			c.nodes[idx].StakingAmount += stakingAmount
+			c.nodeIndex[nodeAddr] = idx // point to existing element
+		}
+	}
+	return c
+}
+
+func (c *ConsolidatedStakingInfo) GetAllNodes() []consolidatedNode {
+	return c.nodes
+}
+
+func (c *ConsolidatedStakingInfo) GetConsolidatedNode(nodeAddr common.Address) *consolidatedNode {
+	if idx, ok := c.nodeIndex[nodeAddr]; ok {
+		return &c.nodes[idx]
+	} else {
+		return nil
+	}
```

Comment:
It is minor, though.

```suggestion
	if idx, ok := c.nodeIndex[nodeAddr]; ok {
		return &c.nodes[idx]
	}
	return nil
```

Reply:
Fixed in fc5dba2e.